tenets/codelingo/code-review-comments/camel-case-constants/codelingo.yaml

```diff
@@ -0,0 +1,55 @@
+funcs:
+  - name: hasUnderscore
+    type: asserter
+    body: |
+      function(varName) {
+        return varName !== "_" && varName.indexOf("_") !== -1 
+      }
+  - name: fixName
+    type: resolver
+    body: |
+      function(varName, private){
+        if(varName.indexOf("_") === -1 || varName === "_") 
+          return varName
+        varName = varName.toLowerCase()
+        pieces = varName.split("_");
+        constName = []
+        if(private == "true"){
+          constName.push(pieces[0])
+        }
+        else{
+          constName.push(pieces[0].charAt(0).toUpperCase() + pieces[0].substring(1))
+        }
+        for (var i = 1; i < pieces.length; i++) {
+          constName.push(pieces[i].charAt(0).toUpperCase() + pieces[i].substring(1))
+        }
+        return constName.join('')
+      }
+
+tenets:
+  - name:  camel-case-constants
+    flows:
+      codelingo/review:
+        comment: Define constant "{{varName}}" as "{{fixName(varName, varPrivate)}}". The convention in Go is to use MixedCaps or mixedCaps rather than underscores to write multiword constant names. [as specified in Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps)
+      codelingo/docs:
+        title: Camel case constants
+        body: |    
+          See https://golang.org/doc/effective_go.html#mixed-caps. This applies even when it breaks conventions in other languages. For example an unexported constant is maxLength not MaxLength or MAX_LENGTH.
+      codelingo/rewrite:
+        #Currently, rewrite is incomplete as all the references to each constant should be fixed too.
+
+      
+    query: |
+      import codelingo/ast/go
+     
+      go.gen_decl(depth = any):
+        tok == "const"
+        go.value_spec:
+          
+          go.names:
+            @review comment
+            @rewrite --replace "{{fixName(varName, varPrivate)}}"
+            go.ident:
+              name as varName
+              hasUnderscore(varName)
```

Comment:
We would also want to catch things like `const VALUE = 5`. Can you please add a check for that too?

Reply:
Maybe only check that on words > 4 in length, so we don't catch acronyms like URL

Reply:
`VALUE` is one word, why would we want to change that? I thought the rule is for multi words.